lib/api/BackbeatAPI.js

```diff
@@ -1335,28 +1307,73 @@ class BackbeatAPI {
      * @returns {undefined} undefined
      */
     _setupMongoClient(cb) {
-        const mongoConfig = this._config.queuePopulator.mongo;
-        const mongoUrl = constructConnectionString(mongoConfig);
-        MongoClient.connect(mongoUrl, {
+        if (this._config.queuePopulator.mongo) {
+            const mongoConfig = this._config.queuePopulator.mongo;
```

Comment:
```suggestion
        const mongoConfig = this._config.queuePopulator.mongo;
        if (!mongoConfig) {
            this._logger.debug('MongoDB configuration not found, skipping MongoDB client setup', {
                method: 'BackbeatAPI._setupMongoClient',
            });
            return cb();
         }
```
Just my opinion, but i think doing it the other way is cleaner ?